### PR TITLE
add docker registry username/password 

### PR DESCRIPTION
**Pull request by pipipipipi43**
#### What type of this PR

Add one of the following kinds:

/kind feature

#### What this PR does / why we need it:

Added docker username and password

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/all?id=210067&issueFilter__urlQuery=eyJpdGVyYXRpb25JRHMiOls1MDZdLCJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMTAwMDgwNSJdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6InRhYmxlIiwiY2hpbGRyZW5WYWx1ZSI6eyJrYW5iYW4iOiJkZWFkbGluZSJ9fQ%3D%3D&iterationID=506&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Support third-party docker registry        |
| 🇨🇳 中文    |   支持第三方 docker registry          |

### CODE REVIEW

**Comment by pipipipipi43** [comment]
related erda pr: https://github.com/erda-project/erda-actions/pull/175

File: modules/pipeline/services/pipelinesvc/secret.go [hunk 1]
```diff
@@ -247,3 +247,10 @@ func (s *PipelineSvc) FetchSecrets(p *spec.Pipeline) (secrets, cmsDiceFiles map[
 
 	return secrets, cmsDiceFiles, holdOnKeys, encryptSecretKeys, nil
 }
+
+func AddRegistryLabel(r map[string]string, clusterInfo apistructs.ClusterInfoData) map[string]string {
```
Comment (kakj-go) [inline modules/pipeline/services/pipelinesvc/secret.go:251]: add method annotation

File: apistructs/cluster_info.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ const (
 	NEXUS_USERNAME          ClusterInfoMapKey = "NEXUS_USERNAME"          // nexus 用户名
 	NEXUS_PASSWORD          ClusterInfoMapKey = "NEXUS_PASSWORD"          // nexus的密码
 	REGISTRY_ADDR           ClusterInfoMapKey = "REGISTRY_ADDR"           // registry的地址
+	REGISTRY_USERNAME       ClusterInfoMapKey = "REGISTRY_USERNAME"       // registry 用户名
```
Comment (sfwn) [inline apistructs/cluster_info.go:45]: english comment

Comment (sfwn) [reply apistructs/cluster_info.go:45]: You should ensure SRE will inject these two fields.
Any related issues?

File: modules/pipeline/services/pipelinesvc/secret.go [hunk 3]
```diff
@@ -135,13 +135,12 @@ func (s *PipelineSvc) FetchPlatformSecrets(p *spec.Pipeline, ignoreKeys []string
...
 
 		// buildpack
-		"bp.repo.prefix":                "", // 兼容用户 pipeline.yml 里写的 ((bp.repo.prefix))
-		"bp.repo.default.version":       "", // 兼容用户 pipeline.yml 里写的 ((bp.repo.default.version))
-		"bp.nexus.url":                  httpclientutil.WrapProto(clusterInfo.Get(apistructs.NEXUS_ADDR)),
-		"bp.nexus.username":             clusterInfo.Get(apistructs.NEXUS_USERNAME),
-		"bp.nexus.password":             clusterInfo.Get(apistructs.NEXUS_PASSWORD),
-		secretKeyDockerArtifactRegistry: httpclientutil.RmProto(clusterInfo.Get(apistructs.REGISTRY_ADDR)),
-		"bp.docker.cache.registry":      httpclientutil.RmProto(clusterInfo.Get(apistructs.REGISTRY_ADDR)),
+		"bp.repo.prefix":           "", // 兼容用户 pipeline.yml 里写的 ((bp.repo.prefix))
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/secret.go:138]: english comment if adjusted.

File: modules/pipeline/services/pipelinesvc/secret.go [hunk 4]
```diff
@@ -247,3 +247,11 @@ func (s *PipelineSvc) FetchSecrets(p *spec.Pipeline) (secrets, cmsDiceFiles map[
 
 	return secrets, cmsDiceFiles, holdOnKeys, encryptSecretKeys, nil
 }
+
+// Support third-party docker registry
+func AddRegistryLabel(r map[string]string, clusterInfo apistructs.ClusterInfoData) map[string]string {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/secret.go:252]: format: AddRegistryLabel xxx
begin with method name.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**